domain/templates/client-with-responses.tmpl

```diff
@@ -1,107 +1,194 @@
-{{range .Bodies}}
-    {{$opid}}{{.Suffix}}WithResponse(ctx context.Context{{genParamArgs $pathParams}}{{if $hasParams}}, params *{{$opid}}Params{{end}}, body {{$opid}}{{.NameTag}}RequestBody) (*{{genResponseTypeName $opid}}, error)
-{{end}}{{/* range .Bodies */}}
-{{end}}{{/* range . $opid := .OperationId */}}
-}
-
-{{range .}}{{$opid := .OperationId}}{{$op := .}}
-type {{$opid | ucFirst}}Response struct {
-    Body         []byte
-	HTTPResponse *http.Response
-    {{- range getResponseTypeDefinitions .}}
-    {{.TypeName}} *{{.Schema.TypeDecl}}
-    {{- end}}
-}
-
-// Status returns HTTPResponse.Status
-func (r {{$opid | ucFirst}}Response) Status() string {
-    if r.HTTPResponse != nil {
-        return r.HTTPResponse.Status
+{{$hasResponse := hasSingle2xxJSONResponse . -}}
+{{$lenServers :=  0 -}}
+{{if .Spec.Servers}}
+{{$lenServers =  len .Spec.Servers -}}
+{{end}}
+{{$response2xx := get2xxResponseTypeDefinition . -}}
+{{$hasBodyOrPathParams := or .HasBody  (gt (len .PathParams) 0) -}}
+
+// {{$opid}} calls the {{.Method}} on {{.Path}}
+// {{.Summary}}
+func (c *Client) {{$opid}}(ctx context.Context, {{if $hasBodyOrPathParams}}params *{{$opid}}AllParams{{else}}{{if $hasParams}} params *{{$opid}}Params{{end}}{{end}}) ({{if $hasResponse}}*{{$response2xx.Schema.TypeDecl}},{{end}} error) {
+    var err error
+{{if .HasBody -}}
+    var bodyReader io.Reader
+    buf, err := json.Marshal(params.Body)
+    if err != nil {
+        return {{if $hasResponse}}nil, {{end}}err
     }
-    return http.StatusText(0)
-}
-
-// StatusCode returns HTTPResponse.StatusCode
-func (r {{$opid | ucFirst}}Response) StatusCode() int {
-    if r.HTTPResponse != nil {
-        return r.HTTPResponse.StatusCode
+    bodyReader = bytes.NewReader(buf)
+{{end}}
+{{range $paramIdx, $param := .PathParams}}
+    var pathParam{{$paramIdx}} string
+    {{if .IsPassThrough}}
+    pathParam{{$paramIdx}} = params.{{.GoVariableName|ucFirst}}
+    {{end}}
+    {{if .IsJson}}
+    var pathParamBuf{{$paramIdx}} []byte
+    pathParamBuf{{$paramIdx}}, err = json.Marshal(params.{{.GoVariableName|ucFirst}})
+    if err != nil {
+        return {{if $hasResponse}}nil, {{end}}err
     }
-    return 0
-}
+    pathParam{{$paramIdx}} = string(pathParamBuf{{$paramIdx}})
+    {{end}}
+    {{if .IsStyled}}
+    pathParam{{$paramIdx}}, err = runtime.StyleParamWithLocation("{{.Style}}", {{.Explode}}, "{{.ParamName}}", runtime.ParamLocationPath, params.{{.GoVariableName|ucFirst}})
+    if err != nil {
+        return {{if $hasResponse}}nil, {{end}}err
+    }
+    {{end}}
 {{end}}
+    serverURL, err := url.Parse(c.{{if eq $lenServers 0}}APIEndpoint{{else}}Server{{end}})
+    if err != nil {
+        return {{if $hasResponse}}nil, {{end}}err
+    }
 
+    operationPath := fmt.Sprintf("{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}})
+    if operationPath[0] == '/' {
+        operationPath = "." + operationPath
+    }
```

Comment:
```suggestion
    operationPath := fmt.Sprintf(".{{genParamFmtString .Path}}"{{range $paramIdx, $param := .PathParams}}, pathParam{{$paramIdx}}{{end}})
```

Reply:
This assumes the `path` parameter of the swagger operation always starts with the leading slash character. Despite it being the best practice, it is not a requirement in the Swagger spec. Can we really rely on it?

Reply:
It appears this way in influxdata swagger files, anyway ... you might be able to use a template `{{if` rather than generated `if`.